src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/TypeImpl.java

```diff
@@ -129,10 +129,10 @@ public int hashCode() {
         @Override
         public boolean equals(Object o) {
             if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
-            DelegatedBase that = (DelegatedBase) o;
-            return kind == that.kind &&
-                    name.equals(that.name);
+            if (!(o instanceof Type.Delegated)) return false;
+            Type.Delegated that = (Type.Delegated) o;
+            return kind == that.kind() &&
+                    name.equals(that.name());
```

Comment:
Delegate is tricky, I notice that we didn't check for type equals.

Just wondering what do we want for equals on multiple-delegated type in different order? Like a unsigned atomic int and a atomic unsigned int?